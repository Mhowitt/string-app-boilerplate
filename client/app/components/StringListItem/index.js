import React from 'react';
import PropTypes from 'prop-types';

const StringListItem = ({ string }) => {
  return (
    <div>
      <div>{string}</div>
    </div>
  );
};

export default StringListItem;

StringListItem.propTypes = {
  string: PropTypes.string,
};
