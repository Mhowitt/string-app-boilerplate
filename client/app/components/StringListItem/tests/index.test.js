import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import StringListItem from '../index';

describe('<StringListItem />', () => {
  let wrapper;

  it('renders without crashing', () => {
    wrapper = shallow(<StringListItem string="test string" />);
  });

  it('renders a <div>', () => {
    wrapper = shallow(<StringListItem string="test string" />);
    expect(wrapper.text()).toEqual('test string');
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
