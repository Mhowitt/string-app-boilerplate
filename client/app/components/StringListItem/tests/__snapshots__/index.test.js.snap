// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StringListItem /> matches snapshot 1`] = `
<div>
  <div>
    test string
  </div>
</div>
`;
